app/Controllers/Http/ProductsController.ts

```diff
@@ -0,0 +1,96 @@
+import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext';
+import Product from 'App/Models/Product';
+import StoreProduct from 'App/Validators/StoreProductValidator';
+import UpdateProduct from 'App/Validators/UpdateProductValidator';
+
+export default class ProductsController {
+    public async index({}: HttpContextContract) {
+        const product = await Product.all();
+
+        return {
+            status: 'success',
+            data: product,
+            message: 'Products retrieved successfully'
+        }
+    }
+
+    public async store({ request, response }: HttpContextContract) {
+        const product = new Product();
+        const payload = await request.validate(StoreProduct);
+
+        product.name = payload.name;
+        product.price = payload.price;
+        await product.save();
+        response.status(201);
```

Comment:
Is it correct to set the status if things are not persisted? Dont understand this.

Also when you return, you dont return anything that has to do with response?

How does that work?

Reply:
When returning something, I always return what I expect such as the data which in this case is a **product** using https://github.com/omniti-labs/jsend. 
JSend is a specification for a simple, no-frills, JSON-based format for application-level communication, and no it is not okay to set the status if things are not persisted, so will clean this up.

Reply:
JSend is nice, but i dont see how you setting status on response proppegates through your code? Or is it some hidde Adonis thing down the line?

Reply:
Adonisjs handles the response through a ResponseContract and If handled well, it would look like this below:
![image](https://user-images.githubusercontent.com/19782721/129271858-e1b0c5bc-c049-4c67-944c-1ba1105a61fb.png)